fvm/environment/parse_restricted_checker.go

```diff
@@ -133,3 +167,54 @@ func parseRestrict4Arg1Ret[
 
 	return callback(arg1, arg2, arg3, arg4)
 }
+
+func parseRestrict6Arg1Ret[
+	Arg1T any,
+	Arg2T any,
+	Arg3T any,
+	Arg4T any,
+	Arg5T any,
+	Arg6T any,
+	RetT any,
+](
+	txnState *state.TransactionState,
+	opCode string,
+	callback func(Arg1T, Arg2T, Arg3T, Arg4T, Arg5T, Arg6T) (RetT, error),
+	arg1 Arg1T,
+	arg2 Arg2T,
+	arg3 Arg3T,
+	arg4 Arg4T,
+	arg5 Arg5T,
+	arg6 Arg6T,
+) (
+	RetT,
+	error,
+) {
+	err := parseRestricted(txnState, opCode)
+	if err != nil {
+		var value RetT
+		return value, err
+	}
+
+	return callback(arg1, arg2, arg3, arg4, arg5, arg6)
+}
+
+func parseRestrict1Arg2Ret[ArgT any, Ret1T any, Ret2T any](
+	txnState *state.TransactionState,
+	opCode string,
+	callback func(ArgT) (Ret1T, Ret2T, error),
+	arg ArgT,
+) (
+	Ret1T,
+	Ret2T,
+	error,
+) {
+	err := parseRestricted(txnState, opCode)
+	if err != nil {
+		var value1 Ret1T
+		var value2 Ret2T
+		return value1, value2, err
+	}
+
+	return callback(arg)
+}
```

Comment:
iirc golang does not have plan to support generic variadic functions. i'm thinking combining https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source and "text/template" may save us some typing here. but just for entertainment purposes, it is too hacky for production.

Reply:
not supporting adhoc (i.e., overloading / variadic) polymorphism is the correct thing to do from the language designer's perspective.  unlike parametric polymorphism, there's no solid theoretical foundation for adhoc polymorphism.

Reply:
this is why i hate C++ ;)